### PR TITLE
ELEC-130: Add CAN messages

### PR DESCRIPTION
**Pull request by karlding**
I initially wanted to make this a Glide dependency in telemetry, but then I realised that it would be kind of weird for the telemetry client to access that vendored Go library. I think it's easier just to make this a submodule in the projects that need it, and then symlink things to the correct relative locations as needed. We could look into making codegen a build step and regenerate code on each build (or add some sort of intelligent caching mechanism down the road)&mdash;but I think symlinks should be fine for now? We could also tag versions as releases, if we don't want **master**. Since we've basically dropped Windows support at this point across our codebase, I don't see the benefit of hacking around symlinks at the moment (although Windows supports hard links via ``mklink``).

This adds v1 of the CAN messages and the associated protobuf message formats. A lot of the CAN message formats are still up in the air, but feel free to discuss.

In addition, this adds codegen templates for C and TypeScript. For Golang, the plan is to simply read the ASCII Protobuf file and create a map of ids. The reason why we aren't doing this for TypeScript at the moment is because that would probably add a [protobuf.js](https://github.com/dcodeIO/ProtoBuf.js/) dependency, and I don't think we'd prefer to add another dependency&mdash;the codegen should be pretty simple, and we can re-evaluate down the road if it becomes an issue.

### CODE REVIEW

File: out/can_msg.h [hunk 1]
```diff
@@ -0,0 +1,47 @@
+
+// For setting the CAN device
+typedef enum {
+
```
Comment (ckitagawa) [inline out/can_msg.h]: What's with this weird newline?

File: out/can_msg.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
+
+// For setting the CAN device
+const enum CanDevice {
```
Comment (ckitagawa) [inline out/can_msg.ts]: Add the `export` keyword to these or they aren't accessible/visible to other modules via require.js

File: can_messages.asciipb [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+    frame {
+      type: UINT64
+    }
+  }
+}
+
+msg {
+  id:23
+  msg_name: "heartbeat response"
+  msg_type: true
```
Comment (ckitagawa) [inline can_messages.asciipb]: Thoughts on turning this field into an enum? This really threw me off until I checked the schema. The issue is it makes a compressed proto 31 bits larger which if we even ship data using will be pretty ugly.

File: out/can_msg.ts (hunk 2 above)
Comment (karlding) [inline out/can_msg.ts]: Ah good catch

File: out/can_msg.h (hunk 1 above)
Comment (karlding) [inline out/can_msg.h]: Fixed :)

File: can_messages.asciipb (hunk 3 above)
Comment (karlding) [inline can_messages.asciipb]: We can simply rename this field to better reflect its purpose&mdash;``is_ack`` or something.

**Pull request merged**